src/main/java/io/asyncer/r2dbc/mysql/MySqlConnection.java

```diff
@@ -303,13 +303,25 @@ public IsolationLevel getTransactionIsolationLevel() {
         return currentLevel;
     }
 
+    /**
+     * Gets session transaction isolation level(Only for testing).
+     *
+     * @return session transaction isolation level.
+     */
+    IsolationLevel getSessionTransactionIsolationLevel() {
+        return sessionLevel;
+    }
+
     @Override
     public Mono<Void> setTransactionIsolationLevel(IsolationLevel isolationLevel) {
         requireNonNull(isolationLevel, "isolationLevel must not be null");
 
         // Set next transaction isolation level.
         return QueryFlow.executeVoid(client, "SET TRANSACTION ISOLATION LEVEL " + isolationLevel.asSql())
```

Comment:
I believe it is necessary to utilize the `SET SESSION TRANSACTION ISOLATION LEVEL` statement to apply this isolation level to all subsequent transactions.

Reply:
Added a corresponding unit test for `SET SESSION TRANSACTION ISOLATION LEVEL`.